### PR TITLE
refactor: changed field to use # to handle counter testcase

### PR DESCRIPTION
**Pull request by Hardikl**
Fix for these logs in CI tests
image

### CODE REVIEW

**Review by cgrinds — DISMISSED** [review]

File: conf/rest/9.12.0/node.yaml [hunk 1]
```diff
@@ -9,7 +9,7 @@ counters:
   - ^controller.failed_fan.message.message            => failed_fan_message
   - ^controller.failed_power_supply.message.message   => failed_power_message
   - ^controller.over_temperature                      => over_temperature
-  - ^ha.partners.0.name                               => ha_partner
+  - ^ha.partners.#.name                               => ha_partner
```
[the remark this answers is not in the document]
Comment (Hardikl) [reply conf/rest/9.12.0/node.yaml:12]: Earlier I thought of adding for `.0.` which would be similar to `.#.`, but given in `#` array when one element will be there gjson treated as normal value only.

File: integration/test/counter_test.go [hunk 2]
```diff
@@ -182,7 +183,11 @@ func readCounters(t *node.Node, path, kind string, countersData map[string][]cou
 					continue
 				}
 				name, _, _, _ := util.ParseMetric(c)
-				nameArray := strings.Split(name, ".#")
+				if strings.Contains(name, ".0") {
+					nameArray = strings.Split(name, ".0")
+				} else {
+					nameArray = strings.Split(name, ".#")
+				}
```
[the remark this answers is not in the document]
Comment (Hardikl) [reply integration/test/counter_test.go]: Updated with regex and unit tests also.

File: pkg/util/util_test.go [hunk 3]
```diff
@@ -148,3 +148,25 @@ func TestVersionAtLeast(t *testing.T) {
...
+		{"cloud_storage.stores.#.cloud_store.name", "cloud_storage.stores"}, // with .#
+		{"abc.o.1.xyz", "abc.o"},                                            // with .o.1
+		{"abc.xyz", "abc.xyz"},                                              // with .xyz
+		{"interfaces.#.ip.address", "interfaces"},                           // with .ip.address
+	}
+
+	for _, tc := range testCases {
+		got := HandleArrayFormat(tc.name)
+		if got != tc.expected {
+			t.Errorf("HandleArrayFormat expected: %s, got :%s", tc.expected, got)
```
Comment (cgrinds) [inline pkg/util/util_test.go:169]: t.Errorf("HandleArrayFormat got :%s want: %s", got, tc.expected)

File: pkg/util/util.go [hunk 4]
```diff
@@ -24,6 +24,8 @@ import (
 	"strings"
 )
 
+var regex = regexp.MustCompile(`^([a-zA-Z][\w.]*)(\.[0-9#])`)
```
Comment (cgrinds) [inline pkg/util/util.go:27]: maybe arrayPrefixRe or arrayRegex might be a better name?

**Pull request merged**